lib/apis/moray.js

```diff
@@ -612,18 +639,24 @@ Moray.prototype.markAsDestroyed = function (vm, callback) {
     });
 };
 
-
-
 /*
- * Returns an object with sort, limit and offset options for moray
+ * Returns a list of objects each representing a sort option to pass
+ * to moray according to the sort parameter "sortparam" and the marker
+ * parameter "markerParam" passed by the HTTP client.
+ * If no sort option needs to be set, an empty array is returned.
  */
-Moray.prototype._addFilterOptions = function (params) {
-    var options = {};
+function _sortOptionsFromSortAndMarkerParams(sortParam, markerParam) {
+    sortParam = sortParam || '';
+    assert.string(sortParam, 'sortParam must be a string');
```

Comment:
Should this assertion not be _before_ we assign to `softParam` when falsey?  If it's optional, perhaps put it first and use the `optionalString()` method of `assert-plus`?